src/main/java/seedu/duke/util/Parser.java

```diff
@@ -185,6 +198,72 @@ private Command prepareAddNote(String userMessage) {
         }
     }
 
+    private String checkBlank(String input, SystemException.ExceptionType e) throws SystemException {
+        if (input.isBlank()) {
+            throw new SystemException(e);
+        } else {
+            return input.trim();
+        }
+    }
+
+    private Command prepareAddEvent(String userMessage) throws SystemException {
+        // add-e eventTitle /t timing /r occurrance /a time before (default same day)
+
+        String title = "";
+        LocalDateTime dateTime = null;
+        boolean toRemind = false;
+        boolean isRecurring = false;
+
+        try {
+            ArrayList<String[]> splitInfo = splitInfoDetails(userMessage);
+            for (String[] infoDetails : splitInfo) {
+                String prefix = infoDetails[0];
+                switch (prefix) {
+                case PREFIX_TITLE:
+                    title = checkBlank(infoDetails[1], SystemException.ExceptionType.EXCEPTION_MISSING_TITLE);
+                    break;
+                case PREFIX_TIMING:
+                    String timingString = checkBlank(infoDetails[1], SystemException.ExceptionType.EXCEPTION_MISSING_TIMING);
+                    dateTime = dateTimeParser(timingString);
+                    break;
+                case PREFIX_REMIND:
+                    toRemind = true;
+                    break;
+                case PREFIX_RECURRING:
+                    isRecurring = true;
+                    break;
+                default:
+                    throw new SystemException(SystemException.ExceptionType.EXCEPTION_WRONG_PREFIX);
+                }
+            }
+
+            if (title.isBlank()) {
+                throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_TITLE);
+            } else if (dateTime == null) {
+                throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_TIMING);
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            return new IncorrectCommand("No description found after a tag!");
+        } catch (SystemException e) {
+            throw e;
```

Comment:
throw new SystemException instead of return a new IncorrectCommand. And can remove the catch for SystemException.